src/System.Memory/tests/Span/CopyTo.cs

```diff
@@ -181,6 +182,56 @@ public static void CopyToShorterArray()
             Assert.Equal<int>(expected, dst);  // CopyTo() checks for sufficient space before doing any copying.
         }
 
+        private static void CopyTo_LargeSize_TestHelper(long BufferSize)
+        {
+            int GuidCount = (int)(BufferSize / Unsafe.SizeOf<Guid>());
+
+            unsafe
+            {
+                bool allocatedFirst = AllocationHelper.TryAllocNative((IntPtr)BufferSize, out IntPtr memBlockFirst);
+                bool allocatedSecond = AllocationHelper.TryAllocNative((IntPtr)BufferSize, out IntPtr memBlockSecond);
+
+                try
+                {
+                    ref Guid memoryFirst = ref Unsafe.AsRef<Guid>(memBlockFirst.ToPointer());
+                    var spanFirst = new Span<Guid>(memBlockFirst.ToPointer(), GuidCount);
+
+                    ref Guid memorySecond = ref Unsafe.AsRef<Guid>(memBlockSecond.ToPointer());
+                    var spanSecond = new Span<Guid>(memBlockSecond.ToPointer(), GuidCount);
+
+                    Guid theGuid = Guid.Parse("900DBAD9-00DB-AD90-00DB-AD900DBADBAD");
+                    for (int count = 0; count < GuidCount; ++count)
+                    {
+                        Unsafe.Add(ref memoryFirst, count) = theGuid;
+                    }
+
+                    spanFirst.CopyTo(spanSecond);
+
+                    for (int count = 0; count < GuidCount; ++count)
+                    {
+                        var guidfirst = Unsafe.Add(ref memoryFirst, count);
+                        var guidSecond = Unsafe.Add(ref memorySecond, count);
+                        Assert.Equal(guidfirst, guidSecond);
+                    }
+                }
+                finally
+                {
+                    if (allocatedFirst) AllocationHelper.ReleaseNative(ref memBlockFirst);
+                    if (allocatedSecond) AllocationHelper.ReleaseNative(ref memBlockSecond);
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
```

Comment:
Linux does deferred memory allocation. The large memory tests were turned off because of that. Otherwise, we were seeing issues where the test allocation would actually succeed, but would die on Linux (via the OOM killer) as soon as we touched the memory block since that is when the kernel would actually satisfy the allocation request.